docs/userguide/networking/dockernetworks.md

```diff
@@ -0,0 +1,368 @@
+
+```
+/ # ping -w3 container2
+PING container2 (172.17.0.3): 56 data bytes
+64 bytes from 172.17.0.3: seq=0 ttl=64 time=0.125 ms
+64 bytes from 172.17.0.3: seq=1 ttl=64 time=0.130 ms
+64 bytes from 172.17.0.3: seq=2 ttl=64 time=0.172 ms
+
+--- container2 ping statistics ---
+3 packets transmitted, 3 packets received, 0% packet loss
+round-trip min/avg/max = 0.125/0.142/0.172 ms
+```
+
+Finally, use the `cat` command to check the `container1` network configuration:
+
+```
+
+/ # cat /etc/hosts
+172.17.0.2      f2870c98fd50
+127.0.0.1       localhost
+::1     localhost ip6-localhost ip6-loopback
+fe00::0 ip6-localnet
+ff00::0 ip6-mcastprefix
+ff02::1 ip6-allnodes
+ff02::2 ip6-allrouters
+172.17.0.2      container1
+172.17.0.2      container1.bridge
+172.17.0.3      container2
+172.17.0.3      container2.bridge
+```
+Exit `container1` or start a new terminal. Then, attach to `container2` and repeat these three commands.
+
+```
+$ sudo docker attach container2
+
+/ # ifconfig
+eth0      Link encap:Ethernet  HWaddr 02:42:AC:11:00:03
+          inet addr:172.17.0.3  Bcast:0.0.0.0  Mask:255.255.0.0
+          inet6 addr: fe80::42:acff:fe11:3/64 Scope:Link
+          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
+          RX packets:8 errors:0 dropped:0 overruns:0 frame:0
+          TX packets:8 errors:0 dropped:0 overruns:0 carrier:0
+          collisions:0 txqueuelen:0
+          RX bytes:648 (648.0 B)  TX bytes:648 (648.0 B)
+
+lo        Link encap:Local Loopback
+          inet addr:127.0.0.1  Mask:255.0.0.0
+          inet6 addr: ::1/128 Scope:Host
+          UP LOOPBACK RUNNING  MTU:65536  Metric:1
+          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
+          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
+          collisions:0 txqueuelen:0
+          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)
+
+/ # ping container1 -w3
+PING container1 (172.17.0.2): 56 data bytes
+64 bytes from 172.17.0.2: seq=0 ttl=64 time=0.277 ms
+64 bytes from 172.17.0.2: seq=1 ttl=64 time=0.179 ms
+64 bytes from 172.17.0.2: seq=2 ttl=64 time=0.130 ms
+--- container1 ping statistics ---
+3 packets transmitted, 3 packets received, 0% packet loss
+round-trip min/avg/max = 0.113/0.174/0.277 ms
+/ # cat /etc/hosts
+172.17.0.3      bda12f892278
+127.0.0.1       localhost
+::1     localhost ip6-localhost ip6-loopback
+fe00::0 ip6-localnet
+ff00::0 ip6-mcastprefix
+ff02::1 ip6-allnodes
+ff02::2 ip6-allrouters
+172.17.0.2      container1
+172.17.0.2      container1.bridge
+172.17.0.3      container2
+172.17.0.3      container2.bridge
+/ #
+
+```
+
+The default `docker0` bridge network supports the use of port mapping and  `docker run --link` to allow communications between containers in the `docker0` network. These techniques are cumbersome to set up and prone to error. While they are still available to you as techniques, it is better to avoid them and define your own networks instead.
```

Comment:
Not sure we should use "cumbersome" etc. here; too opiniated?

We don't have a full alternative to `--link` yet, because the new networking model does not yet support aliases, which is an important feature of `--link` (it acts as an abstraction layer, being able to reference a linked container using a consistent host name, independent of the container name)